mesh/mesh.cpp

```diff
@@ -3535,15 +3573,44 @@ void Mesh::Loader(std::istream &input, int generate_edges,
    // - does not check the orientation of regular and boundary elements
    if (finalize_topo)
    {
-      FinalizeTopology();
+      // don't generate boundary elements for NCMesh, especially in parallel
+      // (FIXME: how can this even work for conforming ParMesh?
+      //  -- we don't want to generate artificial BEs for internal partitions!)
+      bool generate_bdr = (ncmesh == NULL);
+
+      FinalizeTopology(generate_bdr);
```

Comment:
The generate_bdr flag was added by @pazner for refining mesh.  https://github.com/mfem/mfem/pull/963

I suspect we should never generate boundary in the mesh loader, what do you think @pazner?

Reply:
That also sounds reasonable to me — the boundary elements should be created explicitly by the mesh loader, and not by `FinalizeTopology`. However, I'm not very familiar with the mesh loading logic, so I can't say with complete confidence.

Reply:
OK I disabled boundary element generation in any case on load.